srtcore/buffer.cpp

```diff
@@ -871,6 +881,9 @@ int CRcvBuffer::readBufferToFile(fstream& ofs, int len)
 
 void CRcvBuffer::ackData(int len)
 {
+   SRT_ASSERT(len < m_iSize);
```

Comment:
Would be nice to add a comment explaining that these expectations result from a statement that this function should be called with the size precalculated basing on distance between two sequence numbers, one synchronized with the last ACK-ed position and one being the new ACK value earlier checked if it's still within the range of ever added packets to the receiver buffer.

Reply:
BTW. when the type is changed into `size_t` you spare yourself one assert. But this change may bring more other consequences.

Reply:
Changing `len` to `size_t` is quite simple, as `CRcvBuffer::ackData()` is called from only one place. But this will require casting between signed and unsigned back and forward, that is not desired.

Also not sure about a meaningful comment regarding `SRT_ASSERT`. Added a description to the function declaration instead.